### PR TITLE
All investigate logs include ckey of source (if applicable)

### PR DESCRIPTION
**Pull request by Tsar-Salat**
Ports:
- https://github.com/tgstation/tgstation/pull/71833
- https://github.com/tgstation/tgstation/pull/71982

## About The Pull Request
Moves ckey sources to admin_investigate, rather than writing it on every instance we want to log.

Adds more verbose time stamps.

Additionally, I cleaned up singularity_act and documented it, because it was really confusing, and was adding vars just to return them

## Why It's Good For The Game
 Its cleaner this way, less duplicate ckeys, and a few loggings were cleaned.

Dates in addition to times with logs are better for consistency, and external parsers.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

INVESTIGATE_CARGO

image

TGUI's the one panel

image

</details>

## Changelog
:cl: rkz, dragomagol
code: Cleaned up the way ckeys are attributed in investigate logs
code: cleaned up singularity_act and passing gain values
admin: All investigate logs will now contain dates
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 1]
```diff
@@ -684,7 +684,7 @@
 			gain = 100
 		if(mind.assigned_role == JOB_NAME_CLOWN)
 			gain = rand(-1000, 1000)
-	investigate_log("([key_name(src)]) has been consumed by the singularity.", INVESTIGATE_ENGINES) //Oh that's where the clown ended up!
+	usr.investigate_log("has been consumed by the singularity.", INVESTIGATE_ENGINES) //Oh that's where the clown ended up!
```
Comment (itsmeow) [inline code/modules/mob/living/carbon/human/human_defense.dm]: this was inadvertently changed from src

File: code/modules/mob/living/living_defense.dm [hunk 2]
```diff
@@ -403,7 +403,7 @@
 		client.give_award(/datum/award/achievement/misc/singularity_death, client.mob)
 
 
-	investigate_log("([key_name(src)]) has been consumed by the singularity.", INVESTIGATE_ENGINES) //Oh that's where the clown ended up!
+	usr.investigate_log("has been consumed by the singularity.", INVESTIGATE_ENGINES) //Oh that's where the clown ended up!
```
Comment (itsmeow) [inline code/modules/mob/living/living_defense.dm]: this was inadvertently changed from src

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/human/human_defense.dm (hunk 1 above)
Comment (Tsar-Salat) [reply code/modules/mob/living/carbon/human/human_defense.dm:687]: Good catch.

Fixed in 4c1d911

File: code/modules/mob/living/living_defense.dm (hunk 2 above)
Comment (Tsar-Salat) [reply code/modules/mob/living/living_defense.dm]: Good catch. Fixed in 4c1d911

File: code/modules/mob/living/carbon/human/human_defense.dm (hunk 1 above)
Comment (Tsar-Salat) [reply code/modules/mob/living/carbon/human/human_defense.dm]: Actually, why is this even here? Its just duplicate code of /mob/living/singularity_act()

Comment (Tsar-Salat) [reply code/modules/mob/living/carbon/human/human_defense.dm]: Ive retooled it in d6052e8

**Review by itsmeow — APPROVED** [review]

**Pull request merged**